### PR TITLE
1360 Split the infra deploy of API and IHE GW

### PR DESCRIPTION
**Pull request by leite08**
Ref. metriport/metriport-internal#1560

### Dependencies

none

### Description

Split the infra deploy of API and IHE GW

### Testing

- Staging
  - [ ] API infra deployed
  - [ ] IHE GW infra deployed
- Production
  - [ ] API infra deployed
  - [ ] IHE GW infra deployed

- nothing special

### CODE REVIEW

**Review by jonahkaye — APPROVED** [review]
lgtm

File: .github/workflows/_deploy-cdk-ihe-gw.yml [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+# This workflow checks out both the OSS and Internal repos.
+# Because of that, the each step has to indicate the folder in which it should run at - different
+# from regular workflows that only checkout the repo once and run all steps in the same folder.
+
+on:
+  workflow_call:
+    inputs:
+      build_env:
+        description: "the environment we're building to, either 'staging' or 'production' - only apps that have special build scripts need to set this"
```
Comment (Goncharo) [inline .github/workflows/_deploy-cdk-ihe-gw.yml:11]: nit: "the environment we're building for" sounds more correct

File: .github/workflows/_deploy-cdk-ihe-gw.yml [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        required: false
+        type: boolean
+        default: false
+        description: "whether this workflow should checkout internal's develop/master or a feature branch"
+      AWS_REGION:
+        required: true
+        type: string
+      IHE_GW_CONFIG_BUCKET_NAME:
+        required: true
+        type: string
```
Comment (Goncharo) [inline .github/workflows/_deploy-cdk-ihe-gw.yml:33]: shouldn't this be `required: false` for sandbox?

File: .github/workflows/_deploy-cdk-ihe-gw.yml [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+      SENTRY_ORG:
+        required: false
+      DOCKERHUB_USERNAME:
+        required: true
+      DOCKERHUB_TOKEN:
+        required: true
+      IHE_GW_KEYSTORE_STOREPASS:
+        required: true
+      IHE_GW_KEYSTORE_KEYPASS:
+        required: true
```
Comment (Goncharo) [inline .github/workflows/_deploy-cdk-ihe-gw.yml:52]: ditto: shouldn't this be `required: false` for sandbox?

File: .github/workflows/branch-to-staging.yml [hunk 4]
```diff
@@ -46,14 +46,29 @@ jobs:
       IHE_GW_KEYSTORE_STOREPASS: ${{ secrets.IHE_GW_KEYSTORE_STOREPASS_STAGING }}
       IHE_GW_KEYSTORE_KEYPASS: ${{ secrets.IHE_GW_KEYSTORE_KEYPASS_STAGING }}
 
-  infra-and-lambdas:
+  infra-api-lambdas:
```
Comment (Goncharo) [inline .github/workflows/branch-to-staging.yml:49]: nice - this always confused me

**Review by Goncharo — COMMENTED** [review]
left some comments

File: .github/workflows/_deploy-cdk-ihe-gw.yml (hunk 2 above)
Comment (leite08) [reply .github/workflows/_deploy-cdk-ihe-gw.yml:33]: We don't deploy IHE GW on the sandbox

**Review by Goncharo — APPROVED** [review]
:shipit:

**Pull request merged**